packages/cosmic-swingset/src/chain-main.js

```diff
@@ -307,10 +309,16 @@ export default async function main(progname, args, { env, homedir, agcc }) {
   // here so 'sendToChainStorage' can close over the single mutable instance,
   // when we updated the 'portNums.storage' value each time toSwingSet was called.
   async function launchAndInitializeSwingSet(initAction) {
+    const { XSNAP_KEEP_SNAPSHOTS, NODE_HEAP_SNAPSHOTS = -1 } = env;
+
     /** @type {CosmosSwingsetConfig} */
     const swingsetConfig = harden(initAction.resolvedConfig || {});
     validateSwingsetConfig(swingsetConfig);
-    const { slogfile, vatTranscriptRetention } = swingsetConfig;
+    const { slogfile, vatSnapshotRetention, vatTranscriptRetention } =
+      swingsetConfig;
+    const keepSnapshots =
+      vatSnapshotRetention === 'debug' ||
+      (!vatSnapshotRetention && ['1', 'true'].includes(XSNAP_KEEP_SNAPSHOTS));
```

Comment:
I'm not sure how I feel about the env variable "overriding" an explicit `"operational"` retention in the config. I think that's fine, but maybe a comment here making the intention explicit would help.

Reply:
That actually wouldn't happen, because of the `!vatSnapshotRetention` guard. But at any rate, I've updated both `keepSnapshots` and `keepTranscripts` to more clearly apply fallback logic only when the config value is falsy.